---
title: "Data"
bibliography: 
- bibliography.bib
output: 
  prettydoc::html_pretty:
    theme: cayman
    highlight: github
---
<!--
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{microbiome tutorial - data}
  %\usepackage[utf8]{inputenc}
  %\VignetteEncoding{UTF-8}  
-->


## Importing microbiome data in R


### Importing standard formats (CSV, Mothur, BIOM)

The microbiome package has import functions for certain standard data formats for 16S profiling (Simple CSV, Mothur, biom). For details, see the function help. To import these, use:

```{r read_phyloseq, eval=FALSE}
# Import output CSV files generated by write_phyloseq
pseq1 <- read_phyloseq(otu.file, taxonomy.file, metadata.file, type = "simple")

# Import mother .shared and .taxonomy and metadata files
pseq2 <- read_phyloseq(otu.file, taxonomy.file, metadata.file, type = "mothur")

# Import BIOM files
pseq3 <- read_phyloseq(otu.file, taxonomy.file, metadata.file, type = "biom")
```

You can also use additional [import functions](http://joey711.github.io/phyloseq/import-data) from the independent phyloseq R package. 


### Converting data to phyloseq format in R

Alternatively, you can read your data in R (read.table, read.csv or other standard functions) and convert into phyloseq format. The procedure is well explained in the [phyloseq tutorial](http://joey711.github.io/phyloseq/import-data) from the independent phyloseq R package. 

See also examples on [filtering, subsetting and other data processing](Preprocessing.html) for phyloseq objects.


### Adding metadata to phyloseq 

This shows how to add metadata to a phyloseq object. For reproducibility, we just put use the existing metadata, but this could be replaced by another data.frame (samples x fields). 

```{r add_metadata, eval=TRUE}
# Example data
data(dietswap)
pseq <- dietswap

# Pick the existing metadata from a phyloseq object
# (or retrieve this from another source)
df <- meta(pseq)

# Merge the metadata back in the phyloseq object
pseq2 <- merge_phyloseq(pseq, sample_data(df))
```


## Microbiome example data sets

### Intestinal microbiota profiling of 1006 Western adults

[The HITChip Atlas](Atlas.html) data set is available via the microbiome R package in phyloseq format, and via [Data Dryad](http://doi.org/10.5061/dryad.pk75d) in tabular format. This data set from [Lahti et al. Nat. Comm. 5:4344, 2014](http://www.nature.com/ncomms/2014/140708/ncomms5344/full/ncomms5344.html) comes with 130 genus-like taxonomic groups across 1006 western adults with no reported health complications. Some subjects have also short time series. Load the data in R with:

```{r atlasdata}
library(microbiome)
data(atlas1006) 
print(atlas1006)
```


### Diet swap between Rural and Western populations

A two-week diet swap study between western (USA) and traditional
(rural Africa) diets, reported in [O'Keefe et al. Nat. Comm. 6:6342,
2015](http://dx.doi.org/10.1038/ncomms7342). The data is also
available for download from [Data
Dryad](http://dx.doi.org/10.5061/dryad.1mn1n). Load in R with:

```{r dietswap2}
data(dietswap)
print(dietswap)
```


### Dynamics of the human gut microbiome in inflammatory bowel disease

Data set from [Halfvarson et al. Nature Microbiology 2, 2017](http://www.nature.com/articles/nmicrobiol20174) characterizes longitudinal fluctuations of human intestinal microbiota in Inflammatory bowel disease (IBD) subjects and compare it with those of healthy individuals. The authors use Illumina HiSeq 2000 for the V4 region with 515F/806RBC. This data was downloaded from [Qiita](https://qiita.ucsd.edu/study/description/1629) and the Qiita study ID is 1629.
Load the data in R with:

```{r DynamicsIBD}
data(DynamicsIBD)
print(DynamicsIBD)
```

### Intestinal microbiota versus blood metabolites

Data set from [Lahti et al. PeerJ 1:e32,
2013](https://peerj.com/articles/32/) characterizes associations
between human intestinal microbiota and blood serum lipids. Note that
this data set contains an additional data matrix of lipid
species. Load the data in R with:

```{r peerj2}
data(peerj32)
print(names(peerj32))
```
